src/actions/toggleDropdown.ts

```diff
@@ -0,0 +1,45 @@
+import _ from 'lodash'
+import State from '../@types/State'
+import Thunk from '../@types/Thunk'
+import { registerActionMetadata } from '../util/actionMetadata.registry'
+
+type DropdownType = 'colorPicker' | 'letterCase' | 'sortPicker'
+
+type DropdownStateKeys = 'showColorPicker' | 'showLetterCase' | 'showSortPicker'
+
+// Map dropdown types to their state keys
+const DROPDOWN_STATE_KEYS: Record<DropdownType, DropdownStateKeys> = {
+  colorPicker: 'showColorPicker',
+  letterCase: 'showLetterCase',
+  sortPicker: 'showSortPicker',
+}
+
+/**
+ * Toggle a specific dropdown and close all others.
+ * If no dropdown type is provided, all dropdowns will be closed.
+ */
+const toggleDropdown = (state: State, { dropDownType, value }: { dropDownType?: DropdownType; value?: boolean }) => {
```

Comment:
It can be nice in action-creator thunks like this to add some logic that prevents dispatching when the dropdowns are already closed. It's not really a performance concern, but it can avoids firing middleware unnecessarily and can make debugging easier by reducing the amount of clutter in the action history (for example, when using Redux DevTools).

See: 69d84db346d